client/templates/meetingseries/meetingSeriesEdit.js

```diff
@@ -138,14 +139,53 @@ Template.meetingSeriesEdit.events({
             return;
         }
 
+
+        let usersBeforeEdit = this.visibleFor.concat(this.informedUsers);
         let usersWithRolesAfterEdit = Template.instance().userEditConfig.users.find().fetch();
         let allVisiblesArray = [];
         let allInformedArray = [];
         let meetingSeriesId = this._id;
+
+        if (notifyOnRoleChange) {
+            let usersWithRolesAfterEditForEmails = usersWithRolesAfterEdit.slice();
+
+            for (let i in usersBeforeEdit) {
+                let oldUserId = usersBeforeEdit[i];
+                let oldUserWithRole = new UserRoles(oldUserId);
+                let oldUserRole = oldUserWithRole.currentRoleFor(meetingSeriesId);
+                // Search in after edit users whether the users still exists
+                let matchingUser = usersWithRolesAfterEditForEmails.find(function (user) {
+                    return oldUserWithRole._userId === user._idOrg;
+                });
+
+                // If he does not, his role was removed
+                if (matchingUser === undefined) {
+                    Meteor.call('meetingseries.sendRoleChange', oldUserWithRole.getUser()._id, oldUserRole, undefined, meetingSeriesId);
+                } else {
+                    let index = usersWithRolesAfterEditForEmails.indexOf(matchingUser);
+                    let newUserWithRole = new UserRoles(matchingUser._idOrg);
+                    let newUserRole = matchingUser.roles[meetingSeriesId][0];
+
+                    // Roles have changed
+                    if (newUserRole !== oldUserRole) {
+                        Meteor.call('meetingseries.sendRoleChange', newUserWithRole.getUser()._id, oldUserRole, newUserRole, meetingSeriesId);
+                    }
+                    usersWithRolesAfterEditForEmails.splice(index, 1);
+                }
+            }
+            // The remaining users in the after-edit-array -> got added
+            for (let i in usersWithRolesAfterEditForEmails) {
+                let newUser = usersWithRolesAfterEditForEmails[i];
+                let newUserRole = newUser.roles[meetingSeriesId][0];
+                Meteor.call('meetingseries.sendRoleChange', newUser._idOrg, undefined, newUserRole, meetingSeriesId);
+            }
+        }
+
```

Comment:
Please refactor these many lines out to a separate function/method.
This would increase readability of this event handler drastically.